module_fcst_grid_comp.F90

```diff
@@ -0,0 +1,968 @@
+!
+! Add time Attribute to the exportState
+      call ESMF_AttributeAdd(exportState, convention="NetCDF", purpose="FV3", &
+        attrList=(/ "time               ", &
+                      "time:long_name     ", &
+                      "time:units         ", &
+                      "time:cartesian_axis", &
+                      "time:calendar_type ", &
+                      "time:calendar      " /), rc=rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+      call ESMF_AttributeSet(exportState, convention="NetCDF", purpose="FV3", &
+                               name="time", value=real(0,ESMF_KIND_R8), rc=rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+      write(dateSY,'(I4.4)')date_init(1)
+      write(dateSM,'(I2.2)')date_init(2)
+      write(dateSD,'(I2.2)')date_init(3)
+      write(dateSH,'(I2.2)')date_init(4)
+      write(dateSN,'(I2.2)')date_init(5)
+      write(dateSS,'(I2.2)')date_init(6)
+
+      dateS="hours since "//dateSY//'-'//dateSM//'-'//dateSD//' '//dateSH//':'//    &
+            dateSN//":"//dateSS
+      if (mype == 0) write(0,*)'dateS=',trim(dateS),'date_init=',date_init
+
+      call ESMF_AttributeSet(exportState, convention="NetCDF", purpose="FV3", &
+                               name="time:units", value=trim(dateS), rc=rc)
+!                              name="time:units", value="hours since 2016-10-03 00:00:00", rc=rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+      call ESMF_AttributeSet(exportState, convention="NetCDF", purpose="FV3", &
+                               name="time:long_name", value="time", rc=rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+      call ESMF_AttributeSet(exportState, convention="NetCDF", purpose="FV3", &
+                               name="time:cartesian_axis", value="T", rc=rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+      call ESMF_AttributeSet(exportState, convention="NetCDF", purpose="FV3", &
+                               name="time:calendar_type", value=uppercase(trim(calendar)), rc=rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+      call ESMF_AttributeSet(exportState, convention="NetCDF", purpose="FV3", &
+                               name="time:calendar", value=uppercase(trim(calendar)), rc=rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+!
+! Create FieldBundle for Fields that need to be regridded bilinear
+      if( quilting ) then
+
+        do i=1,num_files
+!
+         name_FB = filename_base(i)
+!
+         if( i==1 ) then
+! for dyn
+           name_FB1 = trim(name_FB)//'_bilinear'
```

Comment:
Logic here is opaque on how to determine regridding method.

Reply:
it is predefined that all the atmosphere 3D fields (dyn fields) are using bilinear